src/benchmarks/micro/libraries/Microsoft.Extensions.Configuration/ConfigurationBinderBenchmarks.cs

```diff
@@ -0,0 +1,47 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using BenchmarkDotNet.Attributes;
+using MicroBenchmarks;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+using System.Text.Json;
+
+namespace Microsoft.Extensions.Configuration
+{
+    [BenchmarkCategory(Categories.Libraries)]
+    public class ConfigurationBinderBenchmarks
+    {
+        [Params(32, 64, 128)]
+        public int MySettingsCount { get; set; }
+
+        private IConfiguration _configuration;
+
+        [GlobalSetup]
+        public void GlobalSetup()
+        {
+            var builder = new ConfigurationBuilder();
+            for (int i = 0; i < this.MySettingsCount; i++)
+            {
+                var s = new MySettings
+                {
+                    IdMapping = new Dictionary<string, string> { [i.ToString()] = i.ToString() }
+                };
+                builder.AddJsonStream(new MemoryStream(Encoding.UTF8.GetBytes(JsonSerializer.Serialize(s))));
```

Comment:
Is there a reason to have 128 separate .json files in an app? This seems unusually high.

Reply:
In our production service, the front end service has N microservices as backend, and each microservice use a separate .json file as configure. And each configure file has a lot of keys. We found the service initialization time is O(N^2) and more. That motivates me to benchmark the none-linear time complexity and optimize it.